cgd_tx_eff/src/edu/ohsu/compbio/txeff/tx_eff_hgvs.py

```diff
@@ -45,7 +46,36 @@ def _noneIfEmpty(value: str):
         return None
     return value
 
-def _correct_indel_coords(pos, ref, alt):
+
+class DupHandler:
+    def __init__(self, pysam_file: PysamTxEff, chrom: str, coord: int, alt: str):
+        self.pysam_file = pysam_file
+        self.chrom = chrom
+        self.coord = coord
+        self.alt = alt[1:]
+        self.size = self.pysam_file.size
+        self.seq = self.pysam_file.faidx_query(self.chrom, self.coord)
+        self.fasta_alt = self.seq[self.size + 1:self.size + len(self.alt) + 1]
```

Comment:
- self.alt is defining the alt allele sequence that we are using to search for a duplicate (from the VCF)
- self.size is a constant defined to limit the amount of sequence we look at in the reference FASTA
- self.seq is the actual sequence from the reference, defined based on the position of the alt allele and the constant self.size
- self.fasta_alt provides the sequence directly following the alternate allele, it is trying to match them to see whether the sequence is duplicated

Basically the idea is the move down the reference sequence and keep looking to see whether the given alternate allele is duplicated.  So, if there is an ATGT insertion, this would query the four bases immediately following the ATGT insertion and find out if we once again saw an ATGT.  If so, that would be a duplication.